src/scripts/clipperUI/componentBase.ts

```diff
@@ -67,89 +75,164 @@ export abstract class ComponentBase<TState, TProps> {
-	 * (idea from http://www.paciellogroup.com/blog/2012/04/how-to-remove-css-outlines-in-an-accessible-manner/)
-	 *
-	 * Example use:
-	 *      <a id="myCoolButton" {...this.enableInvoke(this.myButtonHandler, 0)}>Click Me</a>
-	 */
-	public enableInvoke({callback = undefined, tabIndex = 0, args = undefined, idOverride = undefined, setNameForArrowKeyNav = undefined}: EnableInvokeParams) {
-		// Because of the way mithril does the callbacks, we need to rescope it so that "this" points to the class
+		 * Helper which handles tabIndex, clicks, and keyboard navigation for a component that is part of an Aria Set
+		 *
+		 * Also hides the outline if they are using a mouse, but shows it if they are using the keyboard
+		 * (idea from http://www.paciellogroup.com/blog/2012/04/how-to-remove-css-outlines-in-an-accessible-manner/)
+		 *
+		 * Example use:
+		 *      <a id="myCoolButton" {...this.enableInvoke(this.myButtonHandler, 0)}>Click Me</a>
+		 */
+	enableAriaInvoke({callback = undefined, tabIndex = 0, args = undefined, idOverride = undefined, ariaSetName = undefined, ariaSetDirection = undefined}: EnableAria) {
 		if (callback) {
 			callback = callback.bind(this, args);
 		}
 
 		return {
 			onclick: (e: MouseEvent) => {
+				ComponentBase.handleOnClick(e, idOverride, callback);
+			},
+			onkeyup: (e: KeyboardEvent) => {
 				let element = e.currentTarget as HTMLElement;
+				e.preventDefault();
 
-				// Intentionally sending click event before handling the method
-				// TODO replace this comment with a test that validates the call order is correct
-				let id = idOverride ? idOverride : element.id;
+				ComponentBase.handleEnterAndSpaceKey(e, element, callback);
 
-				Clipper.logger.logClickEvent(id);
+				ComponentBase.handleTabKey(e, element);
 
-				if (callback) {
-					callback(e);
-				}
-			},
-			onkeyup: (e: KeyboardEvent) => {
-				let element = e.currentTarget as HTMLElement;
-				if (e.which === Constants.KeyCodes.enter || e.which === Constants.KeyCodes.space) {
-					// Hitting Enter on <a> tags that contains an href automatically fire the click event, so don't do it again
-					if (!(element.tagName === "A" && element.hasAttribute("href"))) {
-						// Intentionally sending click event before handling the method
-						// TODO replace this comment with a test that validates the call order is correct
-						let id = element.id;
+				ComponentBase.handleHomeAndEndButtons(e, ariaSetName, element);
 
-						Clipper.logger.logClickEvent(id);
+				if (element.hasAttribute("data-" + Constants.CustomHtmlAttributes.setNameForArrowKeyNav)) {
+					let posInSet = parseInt(element.getAttribute("aria-posinset"), 10);
 
-						if (callback) {
-							callback(e);
+					if (ariaSetDirection === Constants.AriaNavigation.vertical) {
+						if (e.which === Constants.KeyCodes.up) {
+							if (posInSet === 1) {
+								return;
+							}
+							let nextPosInSet = posInSet - 1;
```

Comment:
Lots of repeated code e.g., between vertical + up arrow, and horizontal + left arrow